_stbt/sqdiff.c

```diff
@@ -0,0 +1,139 @@
+#include <stdint.h>
+#include <assert.h>
+
+enum PixelDepth {
+    PIXEL_DEPTH_U8 = 0,
+    PIXEL_DEPTH_BGR = 1,
+    PIXEL_DEPTH_BGRx = 2,
+    PIXEL_DEPTH_BGRA = 3,
+};
+
+static uint32_t sqdiff_U8(
+    const unsigned char* a, const unsigned char* b,
+    uint16_t len_px);
+static uint32_t sqdiff_BGRx(
+    const unsigned char* t, const unsigned char* f,
+    uint16_t len_px);
+static uint32_t sqdiff_BGRA(
+    uint32_t *count,
+    const unsigned char* t, const unsigned char* f,
+    uint16_t len_px);
+
+typedef struct _SqdiffResult {
+    uint64_t total;
+    uint32_t count;
+} SqdiffResult;
+
+/* Computes the square difference between template t and frame f and counts
+ * the number of pixels not masked.
+ *
+ * color_depth indicates the layout of t in memory:
+ *
+ *                        template layout       frame layout
+ * PIXEL_DEPTH_U8         U8                    U8
+ * PIXEL_DEPTH_BGR        BGR                   BGR
+ * PIXEL_DEPTH_BGRx       BGRx                  BGR
+ * PIXEL_DEPTH_BGRA       BGRA                  BGR
+ *
+ * t_stride and f_stride are the strides between lines measured in bytes for
+ * f and t respectively.
```

Comment:
You got the "f and t respectively" order backwards. :-P

Reply:
Eagle-eye: fixed.